variables.tf

```diff
@@ -0,0 +1,138 @@
+variable "namespace" {
+  description = "Unique namespace name for your resources"
+  type        = string
+}
+
+variable "prefix" {
+  description = "Prefix for your resources"
+  type        = string
+}
+
+variable "location" {
+  description = "The location refers to the physical region or datacenter where your Azure resources are deployed."
+  type        = string
+}
+
+variable "create_resource_group" {
+  description = "The 'create_resource_group' variable is a boolean flag that determines whether to create a resource group (RG) in Azure or use an exsisting one."
+  type        = bool
+  default     = true
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group"
+  type        = string
+}
+
+variable "aks_oidc_issuer_url" {
+  description = "The URL of the OpenID issuer, only applicable when using Azure AD pod identity"
+  type        = string
+}
+
+variable "kubelet_identity" {
+  description = "The kubelet identity to use for the acr pull federation"
+  type        = string
+  nullable    = false
```

Comment:
question: if nullable is false that makes it a required field which means this nullable option doesnt do anything?

Reply:
I think you are right.